Nickname

NoyFlat opened the pull request.

NoyFlat committed:
Add nickname option
Add to loginServlet doPost that gets the user's nickname
Add user id and nickname to datastore
Add hide nickname choice to users that have a nickname

NoyFlat committed:
Add comments display nickname insted of email
Add submit button to nickname form

NoyFlat committed:
Merge branch 'master' into nickname

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -32,20 +37,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+ 
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Entity entity = new Entity("UserInfo", id);
+      entity.setProperty("id", id);
```
Why not just use the primary key? 

When writing the query you can use the reserved property `"__key__",` see:
https://cloud.google.com/datastore/docs/concepts/queries#key_filters

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -32,20 +37,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
-  private String createJson(String isLoggedIn, String url, String email){
+    @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Add some documentation, mention that this is invoked when a user sets a nickname. It took me time to figure it out

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -32,20 +37,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+    
+  }
+
+  private String createJson(String isLoggedIn, String url, String email, String nickname){
```
Optional:
Why not use Gson.toJson

You can create a minimal private sub class here

NoyFlat committed:
Change according to Or's notes
Change filtering the entities by using key insead of user id field
Add inner class NicknameData that holds all the relevent user data for JSON
Change field name

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -92,4 +88,37 @@ public static String getUserNickname(String id) {
+        this.nickname = nickname;
+    }
+
+    private String getLoginStatus(){
+        return this.loginStatus;
+    }
+
+    private String getNickname(){
+        return this.nickname;
+    }
+
+    private String getUrl(){
+        return this.url;
+    }
+
+    private String getEmail(){
+        return this.email;
+    }
```
I don't think the getters are required for Gson to work

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -56,34 +63,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Query query =
         new Query("UserInfo")
-            .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+            .setFilter(new Query.FilterPredicate("__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("UserInfo", id)));
```
You can replace `__key__` with Entity.KEY_RESERVED_PROPERTY

It's better to have less "Magic strings" if possible

orarch commented on portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java:
```diff
@@ -92,4 +88,37 @@ public static String getUserNickname(String id) {
+  private class NicknameData {
+      
+    private String loginStatus;
+    private String url;
```
Any special reason for this to not be of boolean type?

NoyFlat committed:
Change according to Or's notes
Remove useless getters
Replace magic string
Replace loggedIn parameter to boolean and change the name
